File: 6/6-2/Python/TCPserver.py
# UDP通信
import socket
import sys

#接收参数 本地开放端口
if __name__ == '__main__':

    if len(sys.argv) == 2:

        port = int(sys.argv[1])

        server = socket.socket(type=socket.SOCK_STREAM)
        server.bind(('127.0.0.1',port))

        print('Waiting for Connection')
        server.listen(10)
        print('Waiting for Message')
        con,addr = server.accept()
        data = con.recv(1024)
        print('Message recived')
        str = data.decode('utf-8')
        print('Message:'+str)

        print('Sending message')
        str = str.upper()
        con.send(str.encode('utf-8'))
        print('Message sent')

        #con.close()
        server.close()

File: 6/6-2/Java/TCPserver.java
import java.net.ServerSocket;
import java.net.Socket;
import java.io.*;

public class TCPserver {
    public static void main (String[] args) throws Exception {//接收参数 本地开放端口
        byte[] buf = new byte[1024];
        if(args.length != 1)
            return;
        ServerSocket ss = new ServerSocket(Integer.parseInt(args[0]));
        System.out.println("Waiting for connection");
        Socket sk = ss.accept();
        InputStream is = sk.getInputStream();
        int len = is.read(buf);
        System.out.println("Message received");
        String msg = new String(buf,0,len);
        System.out.println("Message:"+msg);
        msg = msg.toUpperCase();

        System.out.println("Sending message");
        OutputStream os = sk.getOutputStream();
        os.write(msg.getBytes());
        System.out.println("Message sent");
        is.close();
        os.close();
        sk.close();
    }
}
